File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.java
import java.util.*;

class Solution {
    public int solution(int[] arr) {
        
        Deque<Integer> que = new ArrayDeque<>();
        
        for(int i=0; i<arr.length; i++){
            que.add(arr[i]);
        }
        
        Arrays.sort(arr);
        
        int i=2;
        int max = arr[arr.length-1];
        int answer = 1;
        
        while(i<=max){
            int cnt = 0;
            int len = que.size();
            for(int j=0; j<len; j++){
                int now = que.poll();
                if(now%i == 0){
                    cnt++;
                    now /= i;
                }
                if(now != 1) que.add(now);
            }
            
            if(cnt == 0) i++;
            else answer *= i;
        }
        
        return answer;
    }
}

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.js
function solution(arr) {
    
    var i = 2;
    arr.sort((a, b) => a - b);
    var max = arr[arr.length-1];
    var answer = 1;
    
    while(i <= max){
        
        var cnt = 0;
        var len = arr.length;
        
        for(let j=0; j<len; j++){
            if(arr[j] == 1) continue;
            if(arr[j]%i == 0){
                cnt++;
                arr[j] /= i; 
            }
        }
        
        if(cnt == 0) i++
        else answer *= i
    }

    return answer;
}